mavros/src/plugins/setpoint_raw.cpp

```diff
@@ -71,6 +77,8 @@ class SetpointRawPlugin : public plugin::PluginBase,
 	ros::Subscriber local_sub, global_sub, attitude_sub;
 	ros::Publisher target_local_pub, target_global_pub, target_attitude_pub;
 
+	std::unique_ptr<double> thrust_scaling_parameter;
```

Comment:
I think that you do not need to dynamically allocate that double. You just need a sign of unset scaling, so NAN is good for that.

Reply:
I'm actually inclined to use `boost::optional`. It would be elegant and does away with dynamic allocs. ROS already depends on boost so the added dependency is not a big issue. What do you think?

Reply:
No. Please avoid Boost.

Reply:
Well, i checked that there are std::optional in C++17, so it's ok to use boost::optional for ROS1. But i would rather limit boost dependency.

Reply:
@vooon Well, I'd hate to let my design choices get in the way of this PR. Just switched to setting a -1.0 sentinel value for `thrust_scaling` and pushed.